src/plugins/expressions/common/expression_functions/specs/math.ts

```diff
@@ -0,0 +1,163 @@
+          sum: 'sum()',
+        },
+      })
+    ),
+  executionFailed: () =>
+    new Error(
+      i18n.translate('expressions.functions.math.executionFailedErrorMessage', {
+        defaultMessage: 'Failed to execute math expression. Check your column names',
+      })
+    ),
+  emptyDatatable: () =>
+    new Error(
+      i18n.translate('expressions.functions.math.emptyDatatableErrorMessage', {
+        defaultMessage: 'Empty datatable',
+      })
+    ),
+};
+
+const fallbackValue = {
+  null: null,
+  zero: 0,
+  false: false,
+} as const;
+
+export const math: ExpressionFunctionDefinition<
+  'math',
+  MathInput,
+  MathArguments,
+  boolean | number | null
+> = {
+  name: 'math',
+  type: undefined,
+  inputTypes: ['number', 'datatable'],
+  help: i18n.translate('expressions.functions.mathHelpText', {
+    defaultMessage:
+      'Interprets a {TINYMATH} math expression using a {TYPE_NUMBER} or {DATATABLE} as {CONTEXT}. ' +
+      'The {DATATABLE} columns are available by their column name. ' +
+      'If the {CONTEXT} is a number it is available as {value}.',
+    values: {
+      TINYMATH,
+      CONTEXT: '_context_',
+      DATATABLE: '`datatable`',
+      value: '`value`',
+      TYPE_NUMBER: '`number`',
+    },
+  }),
+  args: {
+    expression: {
+      aliases: ['_'],
+      types: ['string'],
+      help: i18n.translate('expressions.functions.math.args.expressionHelpText', {
+        defaultMessage: 'An evaluated {TINYMATH} expression. See {TINYMATH_URL}.',
+        values: {
+          TINYMATH,
+          TINYMATH_URL,
+        },
+      }),
+    },
+    onError: {
+      types: ['string'],
+      help: i18n.translate('expressions.functions.math.args.onErrorHelpText', {
+        defaultMessage:
+          "In case the {TINYMATH} evaluation fails or returns NaN, the return value is specified by onError. When `'throw'`, it will throw an exception, terminating expression execution (default).",
+        values: {
+          TINYMATH,
+        },
+      }),
+    },
+  },
+  fn: (input, args) => {
+    const { expression, onError } = args;
+    const onErrorValue = onError ?? 'throw';
+
+    if (!expression || expression.trim() === '') {
+      throw errors.emptyExpression();
+    }
+
+    const mathContext = isDatatable(input)
+      ? pivotObjectArray(
+          input.rows,
+          input.columns.map((col) => col.name)
+        )
+      : { value: input };
+
+    try {
+      const result = evaluate(expression, mathContext);
+      if (Array.isArray(result)) {
+        if (result.length === 1) {
+          return result[0];
+        }
+        throw errors.tooManyResults();
+      }
+      if (isNaN(result)) {
+        // make TS happy
+        if (onErrorValue !== 'throw' && onErrorValue in fallbackValue) {
+          return fallbackValue[onErrorValue];
+        }
+        throw errors.executionFailed();
+      }
+      return result;
+    } catch (e) {
+      if (isDatatable(input) && input.rows.length === 0) {
```

Comment:
The `onError` logic has to happen in the catch handler as well to catch things like divide by zero